frontend/Views/Widgets/ViewGraphedStats/StatusBadge.svelte

```diff
@@ -0,0 +1,18 @@
+<script lang="ts">
+    export let run: any; // The data item passed from PaginatedTable
+    export let variant: "success" | "danger" | "warning" | "info" | "secondary" = "secondary";
+
+    // Get status from the run data
+    $: status = run?.status || "";
+
+    // Auto-determine variant based on status if not provided
+    $: badgeVariant = status === "failed" ? "danger" :
+                      status === "passed" || status === "succeeded" ? "success" :
+                      variant;
+
+    $: displayText = status === "failed" ? "Failed" :
+                     status === "passed" || status === "succeeded" ? "Succeeded" :
+                     status.charAt(0).toUpperCase() + status.slice(1);
+</script>
+
+<span class="badge bg-{badgeVariant}">{displayText}</span>
```

Comment:
I think it would be better to go about this like this to avoid 3 nested reactive declarations and several ternary branches:

```
<span class="badge bg-{badgeVariant}">{subUnderscores(run.status).split(" ").map(v => titleCase(v)).join(" ")}</span>
```
Since `run.status` is guaranteed to be there (as long as run itself is returned) you don't have to make assumptions and can just transform that text directly as for status to bg mapping, I suggest

```jsx
// can be defined in TestStatus.js
const NEM_STATUS_TO_TEST_STATUS = { 
  "skipped": "aborted",
  "succeeded": "passed",
}

class="badge {StatusBackgroundCSSClassMap[NEM_STATUS_TO_TEST_STATUS[run.status] || run.status]}"
```

Reply:
true, I've forgot about this one.
Fixed